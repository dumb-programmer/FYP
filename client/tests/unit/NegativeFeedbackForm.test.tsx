import React from "react";
import { render, screen } from "@testing-library/react";
import { expect, describe, it, vi } from "vitest";
import NegaiveFeedbackForm from "../../src/components/NegativeFeedbackForm";
import userEvent from "@testing-library/user-event";

const id = "negative-feedback-form";
const messageId = "abc";
const onSuccessMock = vi.fn();
const renderComponent = () => render(<NegaiveFeedbackForm id={id} messageId={messageId} onSuccess={onSuccessMock} />);

describe("NegativeFeedback", () => {
    it("renders correctly", () => {
        const { container } = renderComponent();

        expect(container).toMatchSnapshot();
    });

    it("comments input works", async () => {
        renderComponent();

        const input = screen.getByRole("textbox");

        await userEvent.type(input, "hello");

        expect(input.value).toEqual("hello");
    });

    it("comments input works", async () => {
        renderComponent();

        const input = screen.getByRole("textbox");

        await userEvent.type(input, "hello");

        expect(input.value).toEqual("hello");
    });

    it("radio inputs work", async () => {
        renderComponent();

        const radioInputs = screen.getAllByRole("radio");

        for (const radioInput of radioInputs) {
            await userEvent.click(radioInput);

            expect(radioInput).toBeChecked();
        }
    });
})