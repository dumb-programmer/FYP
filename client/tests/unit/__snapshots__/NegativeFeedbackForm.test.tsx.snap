// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NegativeFeedback > renders correctly 1`] = `
<div>
  <form
    class="flex flex-col gap-6"
    id="negative-feedback-form"
  >
    <div
      class="form-control flex flex-wrap gap-4 flex-row text-sm"
    >
      <label
        class="flex items-center gap-2 relative bg-gray-200 p-4 rounded-xl w-fit cursor-pointer hover:bg-gray-300 transition-all bg-slate-800 text-white hover:bg-slate-700"
      >
        <input
          class="radio radio-primary invisible absolute"
          name="category"
          type="radio"
          value="offensive/unsafe"
        />
        offensive/unsafe
      </label>
      <label
        class="flex items-center gap-2 relative bg-gray-200 p-4 rounded-xl w-fit cursor-pointer hover:bg-gray-300 transition-all "
      >
        <input
          class="radio radio-primary invisible absolute"
          name="category"
          type="radio"
          value="not-factually-correct"
        />
        not factually correct
      </label>
      <label
        class="flex items-center gap-2 relative bg-gray-200 p-4 rounded-xl w-fit cursor-pointer hover:bg-gray-300 transition-all "
      >
        <input
          class="radio radio-primary invisible absolute"
          name="category"
          type="radio"
          value="other"
        />
        other
      </label>
    </div>
    <textarea
      class="textarea input-bordered"
      name="comments"
      placeholder="(Optional) Something else you'd like to say"
    />
  </form>
</div>
`;
